### PR TITLE
CP-10986: Return read-caching status on VDI.attach.

### PR DESCRIPTION
**Pull request by siddharthv**
Signed-off-by: Siddharth Vinothkumar <siddharth.vinothkumar@citrix.com>

### CODE REVIEW

File: drivers/blktap2.py [hunk 1]
```diff
@@ -1467,6 +1467,7 @@ def attach(self, sr_uuid, vdi_uuid, writable, activate = False):
         # Return backend/ link
         back_path = self.BackendLink.from_uuid(sr_uuid, vdi_uuid).path()
         struct = { 'params': back_path,
+                   'o_direct': util.read_caching_is_restricted(self._session),
```
Comment (simonjbeaumont) [inline drivers/blktap2.py]: I think this is a race condition. The blktap driver gets hold of this value here and it's this that's passed to the Tapdisk class: https://github.com/siddharthv/sm/blob/CP-10986/drivers/blktap2.py#L1564-L1565.

In between these two places the licensing could have changed meaning the tapdisk is forked with one option but the return data says the opposite.

**Comment by simonjbeaumont** [comment]
Also, we need to ensure that all implementations of the SMAPI returns a sensible default for all SRs. I've just grep'd around and see a few other places where this kind of dictionary is created (obviously without `o_direct`) and returned.

**Comment by siddharthv** [comment]
@simonjbeaumont Ping!

**Comment by franciozzy** [comment]
The patch looks good to me. Ship it!

Reviewed-by: Felipe Franciosi <felipe.franciosi@citrix.com>

**Pull request closed without merging**